### DISCUSSION TITLE
Avoid creating duplicate variables

### DISCUSSION THREAD

**Discussion by adrianbn** [VasilKalchev/LiquidMenu#54] [Q&A] [answered]
I may be doing something wrong with the library but I find that at the moment, in order to edit a `char[]` field I seem to need to create an additional pointer with the sole purpose of passing that pointer to the LiquidLine object. For a couple of fields it's fine, but I have a lot of fields and so I have a lot of additional pointers lying around. 

Is there any way to avoid creating such unnecessary variables? 

Here's an example of what I'm talking about (the variable I wish to remove is `switch_ptr`:

```c++
char switch_status[4] = "Off";
char* switch_ptr = switch;

LiquidLine line3(0, 1, "Enabled: ", switch_ptr);

void toggleEnabled1() {
  schedules[0].enabled = !(schedules[0].enabled);
  if (schedules[0].enabled) {
    strncpy(switch_status, "On", sizeof(switch_status) - 1);
  } else {
    strncpy(switch_status, "Off", sizeof(switch_status) - 1);
  }
}

line3.attach_function(1, toggleEnabled1);
```

Cheers!

**Comment by VasilKalchev** [comment, marked as answer]
Hello adrianbn,

There was a problem with the original way the library handled this functionality. This is now fixed and you can now pass `char switch_status[4]` directly to `LiquidLine`.
